react-native/react/engine/transport.desktop.js

```diff
@@ -14,17 +25,12 @@ export default class DesktopTransport extends BaseTransport {
       console.error('No keybased socket file found!')
     }
 
-    let hooks = null
-    if (connectCallback) {
-      hooks = {connected: connectCallback}
-    }
+    this.net_opts.path = sockfile
 
-    super(
-      {path: sockfile, hooks},
-      null,
-      incomingRPCCallback
-    )
-    this.needsConnect = true
-    this.needsBase64 = false
+    if (sockfile) {
+      super._connect_critical_section(cb)
+    } else {
+      cb(new Error('No socketfile'))
```

Comment:
This is all driven by framed msg pack

On Friday, November 20, 2015, Marco Munizaga <notifications@github.com>
wrote:

> In react-native/react/engine/transport.desktop.js
> <https://github.com/keybase/client/pull/1378#discussion_r45530956>:
>
> >
> > -    super(
> > -      {path: sockfile, hooks},
> > -      null,
> > -      incomingRPCCallback
> > -    )
> > -    this.needsConnect = true
> > -    this.needsBase64 = false
> > +    if (sockfile) {
> > +      super._connect_critical_section(cb)
> > +    } else {
> > +      cb(new Error('No socketfile'))
>
> so whoever calls this knows to retry when it gets an error in the callback?
>
> where does the retry happen?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/keybase/client/pull/1378/files#r45530956>.
>